src/dialogs/copyPasteOwnership.tsx

```diff
@@ -0,0 +1,185 @@
+        }
+    }
+}
+
+function makeCandidatesMap(currentUser: cockpit.UserInfo, cwdInfo: FileInfo, clipboard: ClipboardInfo) {
+    const map: Map<string, string> = new Map();
+    const candidates = get_owner_candidates(currentUser, cwdInfo);
+
+    // also add current ownership if it is same for all files (shallow check)
+    const firstFile = clipboard.files[0];
+    const firstOwnerStr = `${firstFile.user}:${firstFile.group}`;
+    if (clipboard.files.every(file => file.user === firstFile.user && file.group === firstFile.group)) {
+        candidates.add(firstOwnerStr);
+    }
+
+    candidates.forEach(owner => {
+        const split = owner.split(':');
+        let key = owner;
+        if (split.length === 2 && split[0] === split[1]) {
+            key = split[0];
+        }
+        if (owner === firstOwnerStr) {
+            key = `${key} ${_("(original owner)")}`;
+        }
+
+        map.set(key, owner);
+    });
+
+    return map;
+}
+
+const CopyPasteAsOwnerModal = ({
+    clipboard,
+    dialogResult,
+    path,
+} : {
+    clipboard: ClipboardInfo,
+    dialogResult: DialogResult<void>,
+    path: string,
+}) => {
+    const [currentUser, setCurrentUser] = useState<cockpit.UserInfo| undefined>();
+    const [selectedOwner, setSelectedOwner] = useState<string | undefined>();
+    const { cwdInfo, addAlert } = useFilesContext();
+
+    useEffect(() => {
+        cockpit.user().then(user => setCurrentUser(user));
+    }, []);
+
+    let candidatesMap: Map<string, string> = new Map();
+    if (superuser.allowed && currentUser && cwdInfo) {
+        candidatesMap = makeCandidatesMap(currentUser, cwdInfo, clipboard);
+        if (selectedOwner === undefined) {
+            setSelectedOwner(candidatesMap.keys().next().value);
+        }
+    }
+
+    if (selectedOwner === undefined) {
+        return;
+    }
+
+    const selectedVal = candidatesMap.get(selectedOwner);
+    cockpit.assert(selectedVal !== undefined, "New file ownership undefined");
+
+    const modalFooter = (
+        <>
+            <Button
+              variant="primary"
+              onClick={() => {
+                  pasteAsOwner(clipboard, path, selectedVal, addAlert);
+                  dialogResult.resolve();
+              }}
+            >
+                {_("Paste")}
+            </Button>
+            <Button variant="link" onClick={() => dialogResult.resolve()}>{_("Cancel")}</Button>
+        </>
+    );
+
+    return (
+        <Modal
+          id="paste-owner-modal"
+          position="top"
+          title={_("Paste as owner")}
+          isOpen
+          onClose={() => dialogResult.resolve()}
+          variant={ModalVariant.small}
+          footer={modalFooter}
+        >
+            <Form isHorizontal>
+                <TextContent>
+                    <Text>
+                        {_(`Files being pasted have a different owner. By default,
+                            ownership will be changed to match the destination directory.`)}
```

Comment:
Please avoid the line break. This is a translatable string, and gettext/weblate scanners are pretty much guaranteed to mess this up. Also, you must use `"` quotes, not backticks for that to work. `test_js_translatable_strings` in test/common/static-code unfortunately only checks for `'` quotes, not backticks -- I'll teach it to do that.

Note that if this causes a too long line, and you really don't like that: it's acceptable to split this on sentences. It's still weird, as it may happen that one sentence gets translated and the other doesn't, so I don't recommend it. In such cases we just have to accept overlong lines.